### DISCUSSION TITLE
Flare Game Demonstration For Godot 4+

### DISCUSSION THREAD

**Discussion by WithinAmnesia** [flareteam/flare-game#946] [General]
The creative community should have an open source MIT license 2D Flare style Action Role Playing Game (A.R.P.G.) built for the Godot Engine Version 4+.

This is a GitHub for such an undertaking. It uses forever free for everyone open source commercial friendly C.C.0 and M.I.T. License assets. This project is being built to give the world to access to working (a) game demonstration(s); just like Flare and its working demonstration game:
https://github.com/WithinAmnesia/ARPG + https://github.com/WithinAmnesia/ARPG/discussions

The demonstration game should be a template for the community to learn and build from that demonstrates the gameplay, combat and possible features.

**Comment by shinobody** [comment]
As a huge fan of both Godot and Flare I am very excited about that!

**Reply by WithinAmnesia** [reply]
I made some progress on a 2D/3D foundation to run an A.R.P.G. like flare but its using the seamless chunk system like minecraft but it like uses 3D but it has a top down camera and uses 2D sprites too. https://github.com/Zylann/voxelgame/issues/101#issuecomment-1958792464 + https://github.com/WithinAmnesia/ARPG/discussions/16 + https://github.com/Zylann/voxelgame/issues/104

**Reply by Des-Nerger** [reply]
@WithinAmnesia, why have you decided to change the flare-game projection from pseudo-isometric to top-down oblique? Is it because of the libre assets' availability? i.e. isometric ones are a way much harder to find / rerender?

Also, Godot 3.x has much wider portability (i.e. can be run on OpenGL (ES) 2.x GPUs, the Firefox web-browser, etc). Which gets it closer to the flare philosophy (as I see it) of being runnable on very old hardware. Is there a really good reason to strive for Godot 4.x?

**Reply by WithinAmnesia** [reply]
Godot 3.XX is being phased out it has no future moving forward. I'm not an old PC hardware enthusiast either but I want as much wide platform availability that is reasonably possible. Also Godot 4.X works with Firefox and there are many browser game examples of this too. Chrome is slightly better but Firefox still forks fine from my multi-boxing tests with JDungeon. They went great so Godot 4.XX works great for Firefox and Chrome.

The 2D view is due to 2D assets and 2.D is very much less common and if you do 3D its better than 2.5D isometric renders to work with and be productive; think Path of Exile or Pillars of Eternity, or Diablo 2 Resurrected. So the 2D and 2.5D can all be made easier for big scale massive multiplayer games in a 3D setting as it has worked out even to my surprise. As in order get the multiplayer seamless chunk system that worked with Godot 4.2.1.NET+ I had tested literally half a thousand or more demos and examples to use yet in the end basically a Minecraft-like engine called Godot_Voxel and its open source forever free demonstration full multiplayer game called 'Blockygame' won and has the strongest foundation from a technical / mechanic point of view.

I tried to start to make a 2D version of the Blockygame demo with Vector2i's. Yet it was much more reasonable to use 3D and do what Zelda A Link Between Worlds or Doom 1993 does with a 2D/3D hybrid. As with a 3D engine and using 2D billboarded sprites and top down perspective to mimic 2D but it runs on the benefits of a 3D system for developing efficiency. 2D has the biggest library of available free assets from the creative community and it has a super efficient and ease of entry to make new assets and it is very efficient with time to make a lot of assets that can be worth playing. Traditionally 2D games are what most devs seem to gravitate towards when making simple to start games and Itch.io has many 2D independent developer games. Thus I started with 2D assets and forever free open source Godot 4.2.1.NET+ capabilities for efficient developing infrastructure. I wanted to have a proper multiplayer first and foremost so here we are. What are your thoughts and suggestions? Any feedback is welcome.
